ggml/src/ggml-cuda/mmq.cuh

```diff
@@ -761,6 +775,67 @@ template <int mmq_y, bool need_check> static __device__ __forceinline__ void loa
     }
 }
 
+template <int mmq_y, bool need_check>
+static __device__ __forceinline__ void load_tiles_mxfp4_fp4(const char * __restrict__ x,
+                                                            int * __restrict__ x_tile,
+                                                            const int kbx0,
+                                                            const int i_max,
+                                                            const int stride) {
+    constexpr int nwarps = mmq_get_nwarps_device();
+    constexpr int warp_size = ggml_cuda_get_physical_warp_size();
+
+#if defined(BLACKWELL_MMA_AVAILABLE)
+    int *      x_qs = (int *) x_tile;
+    uint32_t * x_sc = (uint32_t *) (x_qs + MMQ_TILE_NE_K);
+
+    const int txi = threadIdx.x;
+
+    // Use all 32 threads: 8 threads per row, process 4 rows per warp per iteration
+    constexpr int threads_per_row = 8;  // 8 blocks per row
+    constexpr int rows_per_warp = warp_size / threads_per_row;  // 4 rows per warp
+    const int kbx = txi % threads_per_row;      // block id 0-7
+    const int row_in_warp = txi / threads_per_row;  // which of the 4 rows this thread handles
+
+#pragma unroll
+    for (int i0 = 0; i0 < mmq_y; i0 += rows_per_warp * nwarps) {
+        int i = i0 + threadIdx.y * rows_per_warp + row_in_warp;
+
+        if (need_check) {
+            i = min(i, i_max);
+        }
+
+        const block_mxfp4 * bxi = (const block_mxfp4 *) x + kbx0 + i * stride + kbx;
+
+        int aux_q4[4];
+        memcpy(aux_q4, bxi->qs, 16);
```

Comment:
Not to say that this should be addressed in this PR, but I think that the `block_mxfp4` datatype has poor alignment. Ideally one would have values that are aligned to at least 4 bytes. It my make sense to convert weights from a format of 32 values + 1 scale to one with 128 values + 4 scales. Or to come up with a better way to load weights with poor alignment (something like load them to shared memory in an aligned way first).